Security.md

```diff
@@ -108,13 +108,76 @@ in-order execution and [post-MVP atomic memory primitives](PostMVP.md#threads).
 Similarly, [side channel attacks][] can occur, such as timing attacks against
 modules. In the future, additional protections may be provided by runtimes or
 the toolchain, such as code diversification or memory randomization (similar to
-[address space layout randomization][] (ASLR)), [bounded pointers][] ("fat"
-pointers), or finer-grained control-flow integrity.
+[address space layout randomization][] (ASLR)), or [bounded pointers][] ("fat"
+pointers).
+
+### Control-Flow Integrity
+The effectiveness of control-flow integrity can be measured based on its
+completeness. Generally, there are three types of control-flow transitions that
+need to be protected:
+  1. Direct function calls,
+  2. Indirect function calls,
+  3. Returns.
+
+Together, (1) and (2) are commonly referred to as "forward-edge", since they
```

Comment:
I just realized that the "generally" above leads to confusion. Add "function-internal control flow" to the list and discuss why wasm doesn't need to secure function-internal control flow.